Assertion error during replace operation

```
>>> import numpy as np
>>> a = np.array([[1,2,3,np.inf],[1,-np.inf,4,5]])
>>> f = dt.Frame(a)
>>> f.replace([-np.inf, np.inf], [np.nan, np.nan])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Assertion 'n > 0' failed in "c/frame/replace.cc", line 566
```

Same error, without numpy:
```
>>> import datatable as dt
>>> import math
>>> fr = dt.Frame([[1,2,3,math.inf], [1,-math.inf,4,5]])
>>> fr.replace([-math.inf, math.inf], math.nan)
>>> fr.replace([-math.inf, math.inf], math.nan)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Assertion 'n > 0' failed in "c/frame/replace.cc", line 566
```

--- a/c/frame/replace.cc
+++ b/c/frame/replace.cc
@@ -414,6 +414,7 @@ void ReplaceAgent::process_bool_column(size_t colidx) {
   int8_t* coldata = col->elements_w();
   size_t n = x_bool.size();
   xassert(n == y_bool.size());
+  if (n == 0) return;
   replace_fw<int8_t>(x_bool.data(), y_bool.data(), nrows, coldata, n);
 }
 
@@ -467,6 +468,7 @@ void ReplaceAgent::process_int_column(size_t colidx) {
   } else {
     size_t n = xfilt.size();
     xassert(n == yfilt.size());
+    if (n == 0) return;
     size_t nrows = static_cast<size_t>(col->nrows);
     T* coldata = col->elements_w();
     replace_fw<T>(xfilt.data(), yfilt.data(), nrows, coldata, n);
@@ -520,6 +522,7 @@ void ReplaceAgent::process_real_column(size_t colidx) {
   } else {
     size_t n = xfilt.size();
     xassert(n == yfilt.size());
+    if (n == 0) return;
     size_t nrows = static_cast<size_t>(col->nrows);
     T* coldata = col->elements_w();
     replace_fw<T>(xfilt.data(), yfilt.data(), nrows, coldata, n);

--- a/tests/munging/test_replace.py
+++ b/tests/munging/test_replace.py
@@ -285,6 +285,22 @@ def test_replace_in_copy():
     assert df2.topython() == [[1, 2, 9], [0.0, 6.6, 7.7], ["A", "-", "C"]]
 
 
+def test_replace_do_nothing():
+    # Check that the case when there is nothing to replace works too
+    df1 = dt.Frame([[1, 2], [3, 4], [5.0, 6.6], [-inf, inf]])
+    df1.replace([-99, -inf, inf], None)
+    df1.internal.check()
+    assert df1.topython() == [[1, 2], [3, 4], [5.0, 6.6], [None, None]]
+
+
+def test_replace_do_nothing2(numpy):
+    np = numpy
+    a = numpy.array([[1, 2, 3, np.inf], [1, -np.inf, 4, 5]])
+    df = dt.Frame(a)
+    df.replace([-np.inf, np.inf], [np.nan, np.nan])
+    df.internal.check()
+    assert df.topython() == [[1., 1.], [2., None], [3., 4.], [None, 5.]]
+
 
 
 #-------------------------------------------------------------------------------

